python-sdk/src/astro/databases/aws/redshift.py

```diff
@@ -68,12 +70,13 @@ class RedshiftDatabase(BaseDatabase):
         FileLocation.S3: "load_s3_file_to_table",
     }
 
-    illegal_column_name_chars: List[str] = ["."]
-    illegal_column_name_chars_replacement: List[str] = ["_"]
+    illegal_column_name_chars: list[str] = ["."]
+    illegal_column_name_chars_replacement: list[str] = ["_"]
 
-    def __init__(self, conn_id: str = DEFAULT_CONN_ID):
+    def __init__(self, conn_id: str = DEFAULT_CONN_ID, table: BaseTable | None = None):
```

Comment:
@pankajastro Just a thought, I think the database is a superset and the table is a subset. For example - One database can be associated with multiple tables, but one table can only be associated with one DB. Based on this train of thought, it feels odd that we are adding tables as part of database creation. I feel this is tight coupling. 

If we need a table for hook creation can we pass the table as a param to self.hook instead? WDYT?

Reply:
Yeah, I agree, but it is a bit difficult to handle it in the current design, I  posted https://github.com/astronomer/astro-sdk/issues/1034#issuecomment-1285647029 some insight